xcrypt.h

```diff
@@ -0,0 +1,37 @@
+/* libxcrypt interfaces for code compatibility.
+
+   Copyright (C) 2018 Björn Esser <besser82@fedoraproject.org>
+
+   Redistribution and use in source and binary forms, with or without
+   modification, are permitted.
+
+   THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+   ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+   IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+   ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+   FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+   DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+   OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+   HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+   LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+   OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+   SUCH DAMAGE.  */
+
+#ifndef _XCRYPT_H
+#define _XCRYPT_H 1
+
+#include <crypt.h>
+
+/* Those are kept for code compatibility with older versions
+   (v3.1.1 and earlier) of libxcrypt.
+   We intentionally define these functions as macros here,
+   since we actually want to link compiled applications against
+   the unprefixed indentical functions defined in <crypt.h>.  */
+#ifndef IN_LIBCRYPT  /* Defined when building libxcrypt. */
+# define xcrypt           crypt
+# define xcrypt_r         crypt_r
+# define xcrypt_gensalt   crypt_gensalt
+# define xcrypt_gensalt_r crypt_gensalt_rn
+#endif
```

Comment:
I presume there exist applications that were including this header?  Also see what I said about using `__REDIRECT` instead of macros, when possible.

Reply:
Same here.